src/reducers/org.js

```diff
@@ -880,6 +886,52 @@ export const updateLogEntryTime = (state, action) => {
   );
 };
 
+export const setSearchFilterInformation = (state, action) => {
+  const { searchFilter, cursorPosition } = action;
+  const headers = state.get('headers');
+  try {
+    const searchFilterExpr = headline_filter_parser.parse(searchFilter);
+    state = state.setIn(['search', 'searchFilterExpr'], searchFilterExpr);
+
+    let filteredHeaders = headers;
+    if (!_.isEmpty(searchFilterExpr)) {
+      filteredHeaders = filteredHeaders.filter(header => {
+        return isMatch(searchFilterExpr)(header);
+      });
+
+      state = state.setIn(['search', 'filteredHeaders'], filteredHeaders);
```

Comment:
@munen Is there a specific reason that you put many lines in the try-catch? Because I'd keep the try-catch only around the one line that I expect to fail (the parser). OK, but doing it this way, we don't need `let` instead of `const`.

Reply:
And I'm tempted to simplify line 898 to `filteredHeaders.filter(isMatch(searchFilterExpr));`

Reply:
The main reason would be that I had to think fast yesterday^^ I wanted the refactoring to be finished and be architecturally sound, so that you’re not blocked to continue(;

There’s certainly room for further improvement in the actual logic implementation, though!

Please feel free to make any changes you want in here. Both your suggestions are absolutely valid! I have full confidence in your general programming and JS skills. It’s clear from your code that you know what you’re doing^^ I just want to lend a helping hand where I can - which is mainly React-Redux architecture and acceptance tests so far. Those are very (front end and framework) specific skills. In anything more general, you’re opinion is likely just as good as mine^^ 

But that you for asking, it’s much appreciated. It’s nice and really refreshing that we can work together collaboratively like this even though it’s the first time we do so. That’s not the standard norm and I’m very grateful and happy for that!